inc/saiswitch.h

```diff
@@ -511,6 +511,9 @@ typedef enum _sai_switch_attr_t
     /* Custom range base value */
     SAI_SWITCH_ATTR_CUSTOM_RANGE_START = 0x10000000,
 
+    /* Enable Broadcom switch shell [bool] (default to 0) */
+    SAI_SWITCH_ATTR_BRCM_SWITCH_SHELL_ENABLE,
```

Comment:
the thread model for shell is different on different platforms. some starts a thread, some starts a process. If you take a look at our syncd code. https://github.com/Azure/sonic-sairedis/blob/master/syncd/syncd.cpp#L1194

Does the calling pattern work for you?